scripts/buildChainsJson.js

```diff
@@ -0,0 +1,164 @@
+const path = require('path');
+const { writeFile } = require('fs/promises');
+const { z } = require('zod');
+const fs = require('fs');
+
+const envSchema = z.object({
+  NOVA_CONFIG_VERSION: z.string(),
+  TELENOVA_CONFIG_VERSION: z.string(),
+});
+const env = envSchema.parse(process.env);
+
+const ASSET_ICONS_DIR = `icons/v1/assets/white`
+const NOVA_CONFIG_VERSION = env.NOVA_CONFIG_VERSION;
+const TELENOVA_CONFIG_VERSION = env.TELENOVA_CONFIG_VERSION;
+const CONFIG_PATH = `chains/${TELENOVA_CONFIG_VERSION}/`;
+const TELENOVA_CONFIG_URL = `https://raw.githubusercontent.com/novasamatech/nova-utils/master/chains/${NOVA_CONFIG_VERSION}/`;
+
+const CHAINS_ENV = ['chains_dev.json', 'chains.json'];
+const ALLOWED_CHAINS = require('./data/allowedChains.json');
+
+async function getDataViaHttp(url, filePath) {
+  try {
+    const response = await fetch(url + filePath);
+    if (!response.ok) {
+      throw new Error(`HTTP error! status: ${response.status}`);
+    }
+    return await response.json();
+  } catch (error) {
+    console.error('🔴 Error in getDataViaHttp: ', error?.message || 'getDataViaHttp failed');
+    process.exit(1);
+  }
+}
+
+function fillAssetData(chain) {
+  const assetsList = [];
+  const allowedAssets = ALLOWED_CHAINS[chain.chainId]?.assets || [];
+  const allowedAssetsMap = new Map(allowedAssets.map(asset => [asset.symbol, asset]));
+
+  for (const asset of chain.assets) {
+    const allowedAsset = allowedAssetsMap.get(asset.symbol);
+    if (!allowedAsset) continue;
+
+    assetsList.push({
+      assetId: asset.assetId,
+      symbol: asset.symbol,
+      precision: asset.precision,
+      type: asset.assetId === 0 ? 'native' : asset.type,
+      priceId: asset.priceId,
+      name: allowedAsset.name,
+      icon: replaceUrl(asset.icon, 'asset', asset.symbol),
+    });
+  }
+  return assetsList;
+}
+
+function getTransformedData(rawData) {
+  const filteredData = rawData.filter(chain => chain.chainId in ALLOWED_CHAINS);
+
+  return filteredData.map((chain, index) => {
+    const assets = fillAssetData(chain);
+    const nodes = chain.nodes
+      .filter(node => !node.url.includes('{'))
+      .map(node => ({ url: node.url, name: node.name }));
+
+    return {
+      name: chain.name,
+      chainIndex: index,
+      addressPrefix: chain.addressPrefix,
+      chainId: `0x${chain.chainId}`,
+      assets,
+      nodes,
+      icon: replaceUrl(chain.icon, 'chain'),
```

Comment:
Let's move it under `chainId` otherwise it looks kind of ragged.
<img width="862" alt="Screenshot 2024-07-17 at 8 28 13 AM" src="https://github.com/user-attachments/assets/cde83f2b-1ab8-4f83-ab96-3704488b4694">